internal/controllers/topology/cluster/reconcile_state_test.go

```diff
@@ -734,6 +734,11 @@ func TestReconcileControlPlaneInfrastructureMachineTemplate(t *testing.T) {
 			for k, v := range tt.want.InfrastructureMachineTemplate.GetLabels() {
 				g.Expect(gotInfrastructureMachineTemplate.GetLabels()).To(HaveKeyWithValue(k, v))
 			}
+
+			// If the template was rotated during the reconcile we want to make sure the old template was deleted.
+			if tt.current.InfrastructureMachineTemplate != nil && tt.current.InfrastructureMachineTemplate.GetName() != tt.desired.InfrastructureMachineTemplate.GetName() {
+				g.Expect(r.Client.Get(ctx, client.ObjectKey{Namespace: tt.current.InfrastructureMachineTemplate.GetNamespace(), Name: tt.current.InfrastructureMachineTemplate.GetName()}, &unstructured.Unstructured{}))
```

Comment:
If I'm not wrong this is missing .To(something)
Also, might be to capture the error and check the error is object not found